docs/guide/renderer/implementing_serverside.md

```diff
@@ -1,9 +1,45 @@
 # Implementing Server-Side UI renderer
 
-Server side renderes can be developed as separate python modules, and plugged into *UI Agent* backend using framework 
-available in [UI Agent Core](../ai_apps_binding/pythonlib.md). 
-It is based on [Stevedore Plugin framework](https://docs.openstack.org/stevedore/latest/index.html).
+Server side renderes can be developed as separate python modules, and plugged into *UI Agent* backend using framework available in [UI Agent Core](../ai_apps_binding/pythonlib.md). It is based on [Stevedore Plugin framework](https://docs.openstack.org/stevedore/latest/index.html).
 
-To implement your renderer, you have to create python module, with this structure.
+Thanks to Stevedore and our comprehensively implemented base classes implementing your own renderer is very simple to start with. You just extend our classes, create the package, add it to Python environment during runtime and Stevedore will pick it up as a valid renderer to use.
 
-*Details comming soon...*
\ No newline at end of file
+To get a very good understanding what it takes to implement your own renderer it is worth going throught he sources of our `next_gen_ui_rhds_renderer` as it's a good example of everything that needs doing. In the next sections of this guide we'll be referencing various pieces of this package to illustrate the approach.
+
+## A step by step guide to create a renderer plugin
+Those instructions will assume you're implementing the plugin as part of our NextGenUI repository fork but it's also possible to implement it completely separate, you'll just have to align the steps with your project structure, build scripts etc.
+
+1. Clone [NextGenUI](https://github.com/RedHat-UX/next-gen-ui-agent) repository or your fork of it.
+2. Create a subdirectory in `./libs` directory that will contain source files for your renderer. By convention we tend to name our renderer packages with the pattenr of `next_gen_ui_******_renderer` with an example of `next_gen_ui_rhds_renderer`. This naming convention has no technical consequence, so it's up to you whether you follow it or not.
```

Comment:
Typo `pattenr` instead of `pattern`

Reply:
Thanks, I forgot Markdown editor in VSCode doesn't highlight those. Actually quite many typos fixed.